SlowCheetah.JDT/JsonDocument.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Sayed Ibrahim Hashimi. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See  License.md file in the project root for full license information.
+
+namespace SlowCheetah.JDT
+{
+    using System;
+    using System.IO;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+
+    /// <summary>
+    /// Represents a JSON file to be transformed
+    /// </summary>
+    public class JsonDocument : IEquatable<JsonDocument>
+    {
+        private readonly string documentPath;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JsonDocument"/> class.
+        /// </summary>
+        /// <param name="filePath">Path to the JSON file</param>
+        public JsonDocument(string filePath)
+        {
+            if (string.IsNullOrEmpty(filePath))
+            {
+                throw new ArgumentNullException(nameof(filePath));
+            }
+
+            this.documentPath = filePath;
+            this.DocumentObject = JsonUtilities.LoadObjectFromFile(this.documentPath);
+        }
+
+        /// <summary>
+        /// Gets the JObject corresponding to the root of the document
+        /// </summary>
+        internal JObject DocumentObject { get; private set; }
```

Comment:
I see no reason for `internal` here. The user can apply the transform, but the only way to get the result is to save it to a file and then re-read it from disk?